Runtime/Core/ApiGatewayManagement/AmazonGameLiftClientWrapper.cs

```diff
@@ -42,17 +56,143 @@ public async Task<SearchGameSessionsResponse> SearchGameSessions(SearchGameSessi
             return await _amazonGameLiftClient.SearchGameSessionsAsync(request);
         }
 
-        public static IAmazonGameLift Create(string localEndpoint)
+        public async Task<DescribeGameSessionsResponse> DescribeGameSessions(DescribeGameSessionsRequest request)
+        {
+            return await _amazonGameLiftClient.DescribeGameSessionsAsync(request);
+        }
+
+        private IAmazonGameLift CreateGameLiftClient(RetriveAwsCredentialsResponse credentialsResponse)
+        {
+            return new AmazonGameLiftClient(credentialsResponse.AccessKey, credentialsResponse.SecretKey);
+        }
+        #endregion
+        
+        /// <summary>
+        /// Server region is code dedicated to Amazon GameLift SDK and AWS SDK calls made by the game server. All of these calls will be done via UI Elements or on Startup. 
+        /// </summary>
+        #region Server
+
+        private async Task<ListLocationsResponse> ListLocations(ListLocationsRequest request)
+        {
+            return await _amazonGameLiftClient.ListLocationsAsync(request);
+        }
+
+        private async Task<CreateLocationResponse> CreateLocation(CreateLocationRequest request)
+        {
+            return await _amazonGameLiftClient.CreateLocationAsync(request);
+        }
+
+        private async Task<RegisterComputeResponse> RegisterCompute(RegisterComputeRequest request)
+        {
+            return await _amazonGameLiftClient.RegisterComputeAsync(request);
+        }
+
+        private async Task<GetComputeAuthTokenResponse> GetComputeAuthToken(GetComputeAuthTokenRequest request)
+        {
+            return await _amazonGameLiftClient.GetComputeAuthTokenAsync(request);
+        }
+
+        private Task<CreateFleetResponse> CreateFleet(CreateFleetRequest request)
         {
-            return new AmazonGameLiftClient(s_dummyAwsAccessKey, s_dummyAwsSecretKey, new AmazonGameLiftConfig
+            return _amazonGameLiftClient.CreateFleetAsync(request);
+        }
+        
+        public async Task<string> GenerateAuthToken(string computeName, string fleetId)
+        {
+            try
+            {
+                var computeAuthTokenRequest = new GetComputeAuthTokenRequest
+                {
+                    ComputeName = computeName,
+                    FleetId = fleetId
+                };
+                var computeAuthTokenResponse = await GetComputeAuthToken(computeAuthTokenRequest);
+                return computeAuthTokenResponse.AuthToken;
+            }
+            catch (Exception e)
             {
-                ServiceURL = localEndpoint
-            });
+                Console.WriteLine(e);
+                throw;
+            }
         }
```

Comment:
Now that I'm taking a second look at this, do we really need an overloaded function for this? Can we simply reuse the GetComputeAuthToken from line 90 instead? The caller can then construct the request however they would like. I'm happy to hear other opinions if you feel strongly otherwise, but it would be nice to keep this entire class consistent

Reply:
I agree that because the user is dealing with the Auth Token they should have full control over it, I shall remove that. 

But for the others, they are UI based, unless we want them to have more direct access to those (which also doesn't sounds like it could be a problem, if they want a more custom experience and don't want to use the UI) This would then I could move these creations of the Requests to the UI scripts and keep this cleaner.